src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Startup.cs

```diff
@@ -75,6 +78,8 @@ public override void ConfigureServices(IServiceCollection services)
                 // Built-in initializers:
                 ServiceDescriptor.Singleton<IPostConfigureOptions<OpenIdConnectOptions>, OpenIdConnectPostConfigureOptions>(),
             });
+
+            services.AddTransient<IConfigureOptions<AzureADSettings>, AzureADSettingsConfiguration>();
```

Comment:
At least for this one, maybe the same for others as twitter ad so on.

In the above lines 70 to 72

    ServiceDescriptor.Transient<IConfigureOptions<AuthenticationOptions>, AzureADOptionsConfiguration>(),
    ServiceDescriptor.Transient<IConfigureOptions<AzureADOptions>, AzureADOptionsConfiguration>(),
    ServiceDescriptor.Transient<IConfigureOptions<PolicySchemeOptions>, AzureADOptionsConfiguration>(),

We are configuring `AuthenticationOptions`, `AzureADOptions` and `PolicySchemeOptions`, done by the same `AzureADOptionsConfiguration`, look at it, it uses values from site settings sections to configure the 3 above options. **This is these options that I think are resolved somewhere through `IOptions<>`.**

**So if `AzureADSettings` is never resolved through e.g. `IOptions<AzureADSettings>`, this is useless, even if you tried it by doing it explicitly for testing.**

What you would have to do instead is to also configure the 3 same above options  `AuthenticationOptions`, `AzureADOptions` and `PolicySchemeOptions`, as done by `AzureADOptionsConfiguration` that uses site settings, but with another implementation using config values. And I think use `IPostConfigureOptions` if the intention is that config values overrides site settings, or still register them after, but I recommend post config to well show the intention.

---

Also maybe check for each value if a config value is provided before overriding those from site settings, or maybe only use a config value if site settings doesn't already provide a value, or for simplicity maybe an option to say if all should be configured from config values or all from site settings indifferently. 

Yes I know it may be more or less complex but here we need to know what we want to do. If we use post configure it means that config values will have an higher precedence, so to work as the regular config stack I recommend to just check: if a config value is provided from config before overriding those from site settings.

Yes the regular config stack work like this, if a config source doesn't provide a value it keeps the possible one provided by a previous source, here site settings becoming another config source. The only difference is that site settings can't have a medium priority, it can only be used before or after the config values which are already the result of all other regular config sources. Here by using post configure to use config values, site settings have the lowest priority.

Hmm, but some people will say that what is in site settings should be used, other will say the same for config values, so maybe need to be discussed, hmm one solution on the fly would be to show any currently used value and make it non editable if different than the value in the database, meaning it is overridden by a config value. But first need to confirm the intention, then will see for editing, maybe in a separate PR.

Oops sorry for the long comment ;)

Also did you see the warning

    // The net5.0 5.0.3 build obsoletes 'AzureADOptions' and 'AzureADDefaults',
    //  'Microsoft.Identity.Web' should be used instead.

I really need to leave ;) will see tomorrow.